src/transformers/pipelines/fill_mask.py

```diff
@@ -93,7 +93,7 @@ def ensure_exactly_one_mask_token(self, model_inputs: GenericTensor):
     def preprocess(self, inputs, return_tensors=None, **preprocess_parameters) -> Dict[str, GenericTensor]:
         if return_tensors is None:
             return_tensors = self.framework
-        model_inputs = self.tokenizer(inputs, return_tensors=return_tensors)
+        model_inputs = self.tokenizer(inputs, return_tensors=return_tensors, **preprocess_parameters)
```

Comment:
Really no, this is going to trip in the long run.. 

We can have `tokenizer_kwargs` which would be a dict, which could be passed as-is, but enabling ALL kwargs to be sent directly to the tokenizer is going to lead to issues, it already has in the past.

`max_length` is both used by the tokenizer and the generate methods for instance and they mean different things.

For generic things:

```python
pipe = pipeline(..., tokenizer_kwargs={"Anything": "goes"})
# or 
pipe = pipeline(..., truncation=True)  # This will mean it cannot ever be used for anything else than the tokenizer, which is likely to be OK
```



Better yet, we could enable specific parameters (which could be top_level if there's no risk).

Reply:
Thank you, makes sense!